### DISCUSSION TITLE
Problem with " Application mnesia exited with reason: stopped"

### DISCUSSION THREAD

**Discussion by hubertoWeb** [rabbitmq/rabbitmq-server#8649] [Other]
### Describe the bug

Hello rabbitmq Team,

I have a problem with rabbitMQ on my local system.

I use docker with:
```
version: "3.6"
services:
  rabbitmq:
    image: rabbitmq:3-management
```

Docker Desktop 4.20.1 (110738)
Docker version 24.0.2, build cb74dfc

Apple Macbook Pro with M1 Chip
Ventura 13.4.

I deleted all containers and images related to rabbitMQ.
Then when I do a:
`docker compose up -d`

I always get this error and rabbitMQ stopps:
```
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:18.984151+00:00 [notice] <0.44.0> Application mnesia exited with reason: stopped
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:20.658039+00:00 [info] <0.230.0> Waiting for Mnesia tables for 30000 ms, 9 retries left
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:20.659274+00:00 [info] <0.230.0> Successfully synced tables from a peer
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:20.662647+00:00 [notice] <0.283.0> Feature flags: attempt to enable `stream_sac_coordinator_unblock_group`...
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:20.793670+00:00 [notice] <0.283.0> Feature flags: `stream_sac_coordinator_unblock_group` enabled
rabbitmq-server-rabbitmq-1  | qemu: uncaught target signal 11 (Segmentation fault) - core dumped
rabbitmq-server-rabbitmq-1  | Segmentation fault
```
In the past it worked, like 5 months ago.

### Reproduction steps

`docker compose up -d`

### Expected behavior

that the container runs and does not exit

### Additional context

 docker compose logs
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:06.946635+00:00 [notice] <0.44.0> Application syslog exited with reason: stopped
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:06.966134+00:00 [notice] <0.230.0> Logging: switching to configured handler(s); following messages may not be visible in this log output
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:06.973002+00:00 [notice] <0.230.0> Logging: configured log handlers are now ACTIVE
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:07.539052+00:00 [info] <0.230.0> ra: starting system quorum_queues
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:07.539748+00:00 [info] <0.230.0> starting Ra system: quorum_queues in directory: /var/lib/rabbitmq/mnesia/rabbit@4fb71bcd203a/quorum/rabbit@4fb71bcd203a
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:07.715984+00:00 [info] <0.261.0> ra system 'quorum_queues' running pre init for 0 registered servers
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:07.749375+00:00 [info] <0.262.0> ra: meta data store initialised for system quorum_queues. 0 record(s) recovered
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:07.786151+00:00 [notice] <0.267.0> WAL: ra_log_wal init, open tbls: ra_log_open_mem_tables, closed tbls: ra_log_closed_mem_tables
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:07.857344+00:00 [info] <0.230.0> ra: starting system coordination
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:07.857635+00:00 [info] <0.230.0> starting Ra system: coordination in directory: /var/lib/rabbitmq/mnesia/rabbit@4fb71bcd203a/coordination/rabbit@4fb71bcd203a
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:07.868808+00:00 [info] <0.274.0> ra system 'coordination' running pre init for 0 registered servers
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:07.874965+00:00 [info] <0.275.0> ra: meta data store initialised for system coordination. 0 record(s) recovered
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:07.875747+00:00 [notice] <0.280.0> WAL: ra_coordination_log_wal init, open tbls: ra_coordination_log_open_mem_tables, closed tbls: ra_coordination_log_closed_mem_tables
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:07.899618+00:00 [info] <0.230.0> 
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:07.899618+00:00 [info] <0.230.0>  Starting RabbitMQ 3.12.0 on Erlang 25.3.2.2 [jit]
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:07.899618+00:00 [info] <0.230.0>  Copyright (c) 2007-2023 VMware, Inc. or its affiliates.
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:07.899618+00:00 [info] <0.230.0>  Licensed under the MPL 2.0. Website: https://rabbitmq.com
rabbitmq-server-rabbitmq-1  | 
rabbitmq-server-rabbitmq-1  |   ##  ##      RabbitMQ 3.12.0
rabbitmq-server-rabbitmq-1  |   ##  ##
rabbitmq-server-rabbitmq-1  |   ##########  Copyright (c) 2007-2023 VMware, Inc. or its affiliates.
rabbitmq-server-rabbitmq-1  |   ######  ##
rabbitmq-server-rabbitmq-1  |   ##########  Licensed under the MPL 2.0. Website: https://rabbitmq.com
rabbitmq-server-rabbitmq-1  | 
rabbitmq-server-rabbitmq-1  |   Erlang:      25.3.2.2 [jit]
rabbitmq-server-rabbitmq-1  |   TLS Library: OpenSSL - OpenSSL 3.1.1 30 May 2023
rabbitmq-server-rabbitmq-1  |   Release series support status: supported
rabbitmq-server-rabbitmq-1  | 
rabbitmq-server-rabbitmq-1  |   Doc guides:  https://rabbitmq.com/documentation.html
rabbitmq-server-rabbitmq-1  |   Support:     https://rabbitmq.com/contact.html
rabbitmq-server-rabbitmq-1  |   Tutorials:   https://rabbitmq.com/getstarted.html
rabbitmq-server-rabbitmq-1  |   Monitoring:  https://rabbitmq.com/monitoring.html
rabbitmq-server-rabbitmq-1  | 
rabbitmq-server-rabbitmq-1  |   Logs: <stdout>
rabbitmq-server-rabbitmq-1  | 
rabbitmq-server-rabbitmq-1  |   Config file(s): /etc/rabbitmq/conf.d/10-defaults.conf
rabbitmq-server-rabbitmq-1  | 
rabbitmq-server-rabbitmq-1  |   Starting broker...2023-06-22 08:12:07.931672+00:00 [info] <0.230.0> 
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:07.931672+00:00 [info] <0.230.0>  node           : rabbit@4fb71bcd203a
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:07.931672+00:00 [info] <0.230.0>  home dir       : /var/lib/rabbitmq
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:07.931672+00:00 [info] <0.230.0>  config file(s) : /etc/rabbitmq/conf.d/10-defaults.conf
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:07.931672+00:00 [info] <0.230.0>  cookie hash    : 44Zvpg2cVp6cjDhbQol+8w==
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:07.931672+00:00 [info] <0.230.0>  log(s)         : <stdout>
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:07.931672+00:00 [info] <0.230.0>  data dir       : /var/lib/rabbitmq/mnesia/rabbit@4fb71bcd203a
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:18.762101+00:00 [info] <0.230.0> Running boot step pre_boot defined by app rabbit
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:18.763087+00:00 [info] <0.230.0> Running boot step rabbit_global_counters defined by app rabbit
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:18.765559+00:00 [info] <0.230.0> Running boot step rabbit_osiris_metrics defined by app rabbit
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:18.766267+00:00 [info] <0.230.0> Running boot step rabbit_core_metrics defined by app rabbit
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:18.768305+00:00 [info] <0.230.0> Running boot step rabbit_alarm defined by app rabbit
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:18.842402+00:00 [info] <0.299.0> Memory high watermark set to 3180 MiB (3335041843 bytes) of 7951 MiB (8337604608 bytes) total
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:18.909670+00:00 [info] <0.301.0> Enabling free disk space monitoring (disk free space: 103757205504, total memory: 8337604608)
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:18.910259+00:00 [info] <0.301.0> Disk free limit set to 50MB
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:18.947319+00:00 [info] <0.230.0> Running boot step code_server_cache defined by app rabbit
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:18.948052+00:00 [info] <0.230.0> Running boot step file_handle_cache defined by app rabbit
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:18.951092+00:00 [info] <0.304.0> Limiting to approx 1048479 file handles (943629 sockets)
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:18.952632+00:00 [info] <0.305.0> FHC read buffering: OFF
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:18.952887+00:00 [info] <0.305.0> FHC write buffering: ON
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:18.959581+00:00 [info] <0.230.0> Running boot step worker_pool defined by app rabbit
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:18.960112+00:00 [info] <0.282.0> Will use 4 processes for default worker pool
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:18.960362+00:00 [info] <0.282.0> Starting worker pool 'worker_pool' with 4 processes in it
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:18.963564+00:00 [info] <0.230.0> Running boot step database defined by app rabbit
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:18.969958+00:00 [info] <0.230.0> Node database directory at /var/lib/rabbitmq/mnesia/rabbit@4fb71bcd203a is empty. Assuming we need to join an existing cluster or initialise from scratch...
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:18.970398+00:00 [info] <0.230.0> Configured peer discovery backend: rabbit_peer_discovery_classic_config
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:18.970722+00:00 [info] <0.230.0> Will try to lock with peer discovery backend rabbit_peer_discovery_classic_config
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:18.971745+00:00 [info] <0.230.0> All discovered existing cluster peers:
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:18.971909+00:00 [info] <0.230.0> Discovered no peer nodes to cluster with. Some discovery backends can filter nodes out based on a readiness criteria. Enabling debug logging might help troubleshoot.
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:18.984151+00:00 [notice] <0.44.0> Application mnesia exited with reason: stopped
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:20.658039+00:00 [info] <0.230.0> Waiting for Mnesia tables for 30000 ms, 9 retries left
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:20.659274+00:00 [info] <0.230.0> Successfully synced tables from a peer
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:20.662647+00:00 [notice] <0.283.0> Feature flags: attempt to enable `stream_sac_coordinator_unblock_group`...
rabbitmq-server-rabbitmq-1  | 2023-06-22 08:12:20.793670+00:00 [notice] <0.283.0> Feature flags: `stream_sac_coordinator_unblock_group` enabled
rabbitmq-server-rabbitmq-1  | qemu: uncaught target signal 11 (Segmentation fault) - core dumped
rabbitmq-server-rabbitmq-1  | Segmentation fault

**Comment by michaelklishin** [comment]
```
rabbitmq-server-rabbitmq-1  | qemu: uncaught target signal 11 (Segmentation fault) - core dumped
rabbitmq-server-rabbitmq-1  | Segmentation fault
```

is pretty specific. Erlang segfaults in your environment. Since this is on 3.12, you can try an earlier version or go straight to 26.0.

RabbitMQ does not directly allocate memory and does not have any native code, so it cannot be responsible for a segmentation fault.

**Reply by hubertoWeb** [reply]
I tried some older versions, but does not work. but thanks

**Comment by michaelklishin** [comment]
If you take the core dump, maybe the maintainers of [erlang/otp](https://github.com/erlang/otp) will be able to make sense of it and investigate. We do not maintain Erlang, at least not to that extent and not for OSS users.
